File: Find-All-Numbers-Disappeared-in-an-Array.py
class Solution:
    def findDisappearedNumbers(self, nums):
        n = len(nums)
        for i in range(n):
            idx = abs(nums[i]) - 1
            nums[idx] = -abs(nums[idx])

        res = []
        for i in range(n):
            if nums[i] > 0:
                res.append(i + 1)
        return res

File: Find-All-Numbers-Disappeared-in-an-Array.cpp
class Solution 
{
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) 
    {
        int n = nums.size();
        for (int i = 0; i < n; i++) 
        {
            int idx = abs(nums[i]) - 1;
            nums[idx] = -abs(nums[idx]);
        }

        vector<int> res;
        for (int i = 0; i < n; i++) 
        {
            if (nums[i] > 0)
                res.push_back(i + 1);
        }
        return res;
    }
};
